lib/TokenManager.js

```diff
@@ -91,17 +86,31 @@ function add(sdk, tokenMgmtRef, storage, key, token) {
   }
   tokenStorage[key] = token;
   storage.setStorage(tokenStorage);
-  setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token);
+  setExpireEventTimeout(tokenMgmtRef, key, token);
+
+  // Remove existing promise key
+  delete tokenMgmtRef.refreshPromise[key];
 }
 
 function get(storage, key) {
   var tokenStorage = storage.getStorage();
   return tokenStorage[key];
 }
 
+function getAsync(sdk, tokenMgmtRef, storage, key) {
+  var token = get(storage, key);
+  if (!token || token.expiresAt * 1000 > Date.now()) {
+    return Q.resolve(token);
+  }
+
+  return tokenMgmtRef.autoRefresh
```

Comment:
Not a blocker, but wrapping the whole thing in a promise wrapper (i.e. with `Q.fcall` or whichever method our version of Q supports) means you don't have to worry about implementation details of how the functions we're calling work - for example, if `remove` is updated in the future to throw an error, this code will no longer work as is.

Update: For example, `setExpireEventTimeout` has a try/catch around `storage.getStorage`, this function does not - do we have the chance of throwing an error here vs. returning a rejected promise?

Reply:
I had some issues with this previously (using an old version of `Q`).

I'm pushing an update where we're wrapping the promise in a `new Q` promise. Looks like this is the approach taken elsewhere:
  - [`http.js`](https://github.com/okta/okta-auth-js/blob/1e3f1003be7fd94476c457ee1dd1a319157ce6b1/lib/http.js#L56)
  - [`token.js`](https://github.com/okta/okta-auth-js/blob/master/lib/token.js#L109)